kroxylicious/src/main/java/io/kroxylicious/proxy/config/tls/KeyProvider.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config.tls;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
+@JsonSubTypes({ @JsonSubTypes.Type(KeyPair.class), @JsonSubTypes.Type(KeyStore.class) })
+public interface KeyProvider {
+
+    SslContextBuilder forServer();
```

Comment:
My first choice here would have been to have the `KeyProvider` return a `KeyManager` and give the concern of making the  `SslContextBuilder` responsibility to another class.  The difficulty faced is that whilst the approach is easy fo rJKS/PKCS12 keystores (you use the classes provided by the java.security package), for PEMs it is harder.  You need code to parse PEMs/deal with encrypted keys etc.    Netty has all this code but unfortunately it is not part of its public API.

So I had a choice:

1.  introduce another dependency to handle PEMs.
2. maintain our own code
3. have the key/trust provide produce or act on the SslContextBuilder.